src/app/containers/PageHandlers/withVariant/index.test.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { useParams, useLocation } from 'react-router-dom';
+import WithVariant from '.';
+import { frontpageRegexPath } from '#app/routes/regex';
+
+jest.mock('react-router-dom', () => ({
+  useParams: jest.fn(),
+  useLocation: jest.fn(),
+  // eslint-disable-next-line react/prop-types
+  Redirect: ({ to: { pathname } }) => <p>You are going to {pathname}</p>,
+}));
+
+describe('WithVariant', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const Component = () => <h1>This is the BBC.</h1>;
+  const WithVariantHOC = WithVariant(Component);
+
+  const getMatchProps = (service, path = null) => ({
+    path: path || frontpageRegexPath,
+    params: {
+      service,
+    },
+  });
+
+  describe('service with no default variant', () => {
+    beforeEach(() => {
+      const service = 'news';
+
+      useParams.mockReturnValue({
+        service,
+      });
+      useLocation.mockReturnValue({
+        pathname: `/${service}`,
+      });
+    });
+
+    const match = getMatchProps('news');
+
+    shouldMatchSnapshot(
+      'should not redirect',
+      <WithVariantHOC match={match} />,
+    );
+  });
+
+  describe('service (ukchina) with default variant', () => {
+    beforeEach(() => {
+      const service = 'ukchina';
+
+      useParams.mockReturnValue({
+        service,
+      });
+      useLocation.mockReturnValue({
+        pathname: `/${service}`,
+      });
+    });
+
+    const match = getMatchProps('ukchina');
+
+    shouldMatchSnapshot(
+      'should redirect to */trad',
+      <WithVariantHOC match={match} />,
+    );
+  });
+
+  describe('service (zhongwen) with default variant', () => {
+    beforeEach(() => {
```

Comment:
do we need all the beforeEaches?

Reply:
Yes. It sets the mocks to the expected values. Is there a reason you'd prefer something else?

Reply:
but there's only 1 test in each describe so can you not just remove the beforeEaches and everything will still work as expected?

i.e is this:

```js
describe('service (zhongwen) with default variant', () => {
  const service = 'zhongwen';

  useParams.mockReturnValue({
    service,
  });
  useLocation.mockReturnValue({
    pathname: `/${service}`,
  });

  const match = getMatchProps('zhongwen');

  shouldMatchSnapshot(
    'should redirect to */trad',
    <WithVariantHOC match={match} />,
  );
});
```

equal to this?:

```js
describe('service (zhongwen) with default variant', () => {
  beforeEach(() => {
    const service = 'zhongwen';

    useParams.mockReturnValue({
      service,
    });
    useLocation.mockReturnValue({
      pathname: `/${service}`,
    });
  });

  const match = getMatchProps('zhongwen');

  shouldMatchSnapshot(
    'should redirect to */trad',
    <WithVariantHOC match={match} />,
  );
});

```

Reply:
In this paradigm,

```
describe('service (zhongwen) with default variant', () => {
  const service = 'zhongwen';

  useParams.mockReturnValue({
    service,
  });
  useLocation.mockReturnValue({
    pathname: `/${service}`,
  });

  const match = getMatchProps('zhongwen');

  shouldMatchSnapshot(
    'should redirect to */trad',
    <WithVariantHOC match={match} />,
  );
});
```

`shouldMatchSnapshot` actually contains an `it(...)` function, and because `useParams` and `useLocation` are global variables, there are conflicts because of the way jest runs tests.

Following [this comment](https://github.com/bbc/simorgh/pull/4567#discussion_r345118721), the code has been modified to use the traditional it(...) function, so it's no longer necessary to have the `beforeEach`.

Reply:
Fixed!